lib/resty/iprepd.lua

```diff
@@ -191,17 +199,28 @@ end
 
 function _M.audit_log(self, ip)
   if self.audit_blocked_requests == 1 then
-    ngx.req.read_body()
-    local headers = ngx.req.get_headers()
-    local request_headers_all = ""
-    for k,v in pairs(headers) do
-      request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+    local should_audit_request = 0
+    for i, uri in ipairs(self.audit_uri_list) do
+      if string.match(uri, ngx.var.uri) then
+        should_audit_request = 1
+        break;
+      end
     end
-    local data = ngx.req.get_body_data()
-    if not data then
-      data = ngx.req.get_body_file()
+    if should_audit_request == 1 then
+      ngx.req.read_body()
+      local request_headers_all = ""
+      if self.audit_include_headers == 1 then
+        local headers = ngx.req.get_headers()
+        for k,v in pairs(headers) do
+          request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+        end
+      end
+      local data = ngx.req.get_body_data()
+      if not data then
+        data = ngx.req.get_body_file()
+      end
+      ngx.log(ngx.ALERT, string.format('FoxSec Audit || "%s" || %s || "%s" || "%s" ||', ip, ngx.var.uri, request_headers_all, data))
```

Comment:
How will this handle line feeds/control characters/etc that are present in the request body (or potentially headers)?

Should we encode this string (e.g., base64) prior to logging it to ensure there are no characters in here that could result in problems reading/processing/forwarding the log stream (or otherwise being able to tell where a single event begins/ends)?

Reply:
i added base64 for logging headers and for logging body data.
body filename or msg indicating reading body failed are not.